### ISSUE TITLE
ONNX current dev status

### ISSUE THREAD

**Issue by ahmedhosny** [onnx/onnx#199]
I've been looking around for a good consolidated list of current onnx integrations with different platforms - a concise matrix perhaps that shows what is currently supported and how to convert to/from onnx as well as what operations are supported within each platform.

I put [this](https://github.com/ahmedhosny/modelhub/wiki/Contributing-models-to-modelhub) together for an effort to build a collections of models all formatted in onnx. We will be asking contributors to convert their models and want to be somewhat confident of the conversion protocols.

1. Could you guys give me quick feedback if this is truly representative of the current status?
2. I am happy to start something like this but more detailed (with list of supported operations) perhaps in the onnx docs or in the wiki. Any ideas?

**Comment by dzhulgakov** [comment]
Regarding the list of frameworks that can export ONNX models your list is correct: Caffe/Caffe2, LuaTorch/PyTorch, CNTK. MXNet support is likely coming up soon. Similarly community is building TF support (though it's importer, not export first for both). Indeed, representing it more prominently on the main page is desired and we're going to do it in the next few weeks.

As for operator-level support, we're currently building some test harness to programmatically generate table with support level of each of the operators across backends. It'd tell us programmatically which operators are supported.

For the importing functionality (from framework to onnx) it's a bit more tricky as the input apis for different frameworks are different. The best setup would probably be to build up a collection of models that are converted correctly.

For your project - please do try to use ONNX and provide feedback and missing features here. We're very interested in your use cases.

**Comment by ahmedhosny** [comment]
Thanks @dzhulgakov  - Will definitely provide feedback here. Happy to contribute to the docs as it would help understand why some converted models fail the checks.

**Comment by nouiz** [comment]
For the importer fonctionality, it won't be complete, but it should be
possible to take the exporter tests and after export, test the import of
the just exported model? With that, we could also check in information was
lost in the translation.

**Comment by dzhulgakov** [comment]
Yes, that works, but in general framework's frontend might generate slightly different operator combinations from the exporter. So it won't be a comprehensive test.
